content/docs/id-tokens.md

```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
 
 For details on how to request or validate an ID Token, see [“_Writing apps that use Dex_.”][using-dex]
 
+## Refresh tokens
+Refresh tokens are credentials used to obtain access tokens.  Refresh tokens are issued to the client by the authorization server and are used to obtain 
+a new id token when the current id token becomes invalid or expires. Issuing a refresh token is optional and provided by passing `offline_access` scope to Dex server.
+
+__NOTE__: Some connectors do not support `offline_access` scope. You can find out which connectors support refresh tokens by looking into the [_connectors list_][connectors].
+
+Example of a server response with refresh token:
+```json
+{
+ "access_token": "eyJhbGciOiJSUzI1N...",
+ "token_type": "Bearer",
+ "refresh_token": "lxzzsvasxho5exvwkfa5zhefl",
+ "expires_in": 3600,
+ "id_token": "eyJhbGciO..."
+}
+```
+
+__NOTE__: For every refresh of id token, Dex issues a new refresh token. This security measure is called refresh token rotation
+and prevents stealing it. The idea is described in detail in the corresponding [RFC][rfc6819-5.2.2.3].
+
+## Expiration and rotation settings
+
+Dex has a section in the config file where you can specify expiration and rotation settings for id tokens and refresh tokens.
+__NOTE__: All duration options should be set in the format: number + time unit (s, m, h), e.g., `10m`.
+
+* `expire` - section for various expiration settings, including token settings:
+  * `idTokens` - the lifetime of id_token. It is preferable to use short-lived id tokens.
+  * `refreshTokens` - section for various refresh token settings:
+    * `validForIfNotUsed` - invalidate a refresh token if it is not used for specified amount of time.
+    * `absoluteLifetime` - a stricter variant of the previous option, absolute lifetime of a refresh token. It forces users to reauthenticate and obtain a new refresh token.
+    * `disableRotation` - completely disables every-request rotation. The user will also have to specify one of the previous refresh token options to keep refresh tokens secure by toggling this.
+    * `reuseInterval` - allows getting the same refresh token from refresh endpoint within a specified interval, but only if the user's request contains the previous refresh token. 
+      
+__NOTE__: `disableRotation` and `reuseInterval` options help effectively deal with network lags, concurrent requests, and so on in tradeoff for security. Use them with caution.
+  
+
 [jwt-io]: https://jwt.io/
 [standard-claims]: https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
 [kubernetes]: http://kubernetes.io/docs/admin/authentication/#openid-connect-tokens
 [aws-sts]: https://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html
 [using-dex]: using-dex.md
+[connectors]: connectors/_index.md
```

Comment:
We only need `connectors/` here I beleive, no need for the `_index.md` as the index files get resolved to the folder. Same for `[using-dex]` above.

Reply:
Right: drop `/_index.md` and, in general,`.md` suffixes.